br/pkg/lightning/backend/external/onefile_writer.go

```diff
@@ -0,0 +1,240 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package external
+
+import (
+	"context"
+	"encoding/binary"
+	"encoding/hex"
+	"path/filepath"
+	"strconv"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/br/pkg/lightning/common"
+	"github.com/pingcap/tidb/br/pkg/storage"
+	tidbkv "github.com/pingcap/tidb/pkg/kv"
+	"github.com/pingcap/tidb/pkg/util/logutil"
+	"github.com/pingcap/tidb/pkg/util/size"
+	"go.uber.org/zap"
+)
+
+// OneFileWriter is used to write data into external storage with one file.
+type OneFileWriter struct {
+	store          storage.ExternalStorage
+	writerID       string
+	filenamePrefix string
+	keyAdapter     common.KeyAdapter
+
+	kvStore *KeyValueStore
+	rc      *rangePropertiesCollector
+
+	kvBuffer    *preAllocKVBuf
+	kvLocations []kvLocation
+	kvSize      int64
+
+	onClose OnCloseFunc
+	closed  bool
+
+	// Statistic information per batch.
+	batchSize uint64
+
+	multiFileStat MultipleFilesStat
+	// Statistic information per writer.
+	minKey    tidbkv.Key
+	maxKey    tidbkv.Key
+	totalSize uint64
+
+	dataFile   string
+	statFile   string
+	dataWriter storage.ExternalFileWriter
+	statWriter storage.ExternalFileWriter
+	logger     *zap.Logger
+}
+
+// initWriter inits the underlying dataFile/statFile path, dataWriter/statWriter for OneFileWriter.
+func (w *OneFileWriter) initWriter(ctx context.Context, partSize int64) (
+	err error,
+) {
+	w.dataFile = filepath.Join(w.filenamePrefix, strconv.Itoa(0))
```

Comment:
```suggestion
	w.dataFile = filepath.Join(w.filenamePrefix, "0")
```

or maybe we can change the suffix to "one-file" to distinguish it more obviously